### PR TITLE
INT-6964 - On-prem troubleshooting

### PR DESCRIPTION
**Pull request by adam-in-ict**
Adding an additional CLI command for on-prem troubleshooting:

```
Usage: j1-integration troubleshoot [options]

troubleshoot common issues with local execution

Options:
  -p, --project-path <directory>  path to integration project directory (default: "/Users/apierson/git/graph-sentry")
  -h, --help                      display help for command
```

### CODE REVIEW

File: packages/integration-sdk-cli/src/commands/troubleshoot.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { createCommand } from 'commander';
+// import path from 'path';
```
Comment (zemberdotnet) [inline packages/integration-sdk-cli/src/commands/troubleshoot.ts:2]:
```suggestion
```

File: packages/integration-sdk-cli/src/commands/troubleshoot.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export function troubleshootLocalExecution() {
+  return createCommand('troubleshoot')
+    .description('troubleshoot common issues with local execution')
+    .option(
+      '-p, --project-path <directory>',
+      'path to integration project directory',
+      process.cwd(),
+    )
+    .action(async (options) => {
+      // const integrationDir = path.resolve(options.projectPath);
```
Comment (zemberdotnet) [inline packages/integration-sdk-cli/src/commands/troubleshoot.ts:15]:
```suggestion
```

File: packages/integration-sdk-cli/src/troubleshoot/utils.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  log.info(`Checking availability of endpoint ${url}`);
+  try {
+    urlExists(url, function (err, exists) {
+      if (exists) {
+        log.info(`Endpoint can be successfully reached.`);
+      } else {
+        log.info(
+          `Checking availability of endpoint with certificate verification disabled.`,
+        );
+        process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
```
Comment (zemberdotnet) [inline packages/integration-sdk-cli/src/troubleshoot/utils.ts:63]: Perhaps after we make the call we re-enable certificate validation.

**Review by zemberdotnet — APPROVED** [review]

File: packages/integration-sdk-cli/src/troubleshoot/error.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { IntegrationError } from '@jupiterone/integration-sdk-core';
+
+export class IntegrationMissingValidateFunction extends IntegrationError {
+  constructor(message: string) {
+    super({
+      code: 'MISSING_VALIDATE_FUNCTION',
+      message,
+    });
+  }
+}
```
[the remark this answers is not in the document]
Comment (adam-in-ict) [reply packages/integration-sdk-cli/src/troubleshoot/error.ts:10]: Nope.  I'd intended to remove it after refactoring the validateAuthentication test.  Thanks!

**Pull request merged**